services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3TransferManagerTest.java

```diff
@@ -332,7 +338,7 @@ void downloadDirectory_throwException_shouldCompleteFutureExceptionally() {
     void close_shouldCloseUnderlyingResources() {
         S3TransferManager transferManager = new DefaultS3TransferManager(mockS3Crt, uploadDirectoryHelper, configuration, downloadDirectoryHelper);
```

Comment:
Can we change to use builder pattern and pass the mockS3Crt client so that we can verify that the provided client is not closed when we close the transfer manager?

Reply:
added a new test case with Builder